ray_on_golem/client/client.py

```diff
@@ -79,16 +80,18 @@ def get_cluster_state(self) -> Dict[models.NodeId, models.NodeData]:
         return response.cluster_data
 
     def non_terminated_nodes(self, tag_filters: models.Tags) -> List[models.NodeId]:
-        response = self._make_request(
-            url=settings.URL_NON_TERMINATED_NODES,
-            request_data=models.NonTerminatedNodesRequestData(
-                tags=tag_filters,
-            ),
-            response_model=models.NonTerminatedNodesResponseData,
-            error_message="Couldn't get non terminated nodes",
-        )
-
-        return response.nodes_ids
+        try:
+            response = self._make_request(
+                url=settings.URL_NON_TERMINATED_NODES,
+                request_data=models.NonTerminatedNodesRequestData(
+                    tags=tag_filters,
+                ),
+                response_model=models.NonTerminatedNodesResponseData,
+                error_message="Couldn't get non terminated nodes",
+            )
+            return response.nodes_ids
+        except RayOnGolemClientError:
+            return list()
```

Comment:
Returning an empty collection on error without any additional condition can be misleading, as it could be called on the cluster with no connection to the webserver - it will impact on autoscaler behavior. This change should be removed, or it could be used, but only when server is shutting down, as our code guarantees that the node is not running after completing `provier.terminate_node()`. In the second case, removing shutdown delays should be removed.